File: 682-baseball-game/baseball-game.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        score = []
        for each in operations:
            if each == "C":
                score.pop()
            elif each == "D":
                score.append(score[-1]*2)
            elif each == "+":
                score.append(score[-1] + score[-2])
            else:
                score.append(int(each))
    
        return sum(score)

File: 682-baseball-game/baseball-game.js
/**
 * @param {string[]} operations
 * @return {number}
 */
var calPoints = function(operations) {
    let score = [];
    for (let each of operations){
        if (each == "+")
        score.push(score[score.length-1]+score[score.length-2]);
        else if (each == "D")
        score.push(score[score.length-1]*2);
        else if (each == "C")
        score.pop();
        else
        score.push(parseInt(each))
    }
    let sum = score.reduce((sum, num) => sum + num, 0);
    return sum;
};